src/services/dataRenderService.ts

```diff
@@ -630,12 +664,13 @@ export class DataRenderService {
         lines
             .attr("points", (d) => {
                 const textPoint = [d.labelX, d.labelY];
-                textPoint[0] = textPoint[0] + ((midAngle(<any>d) < Math.PI ? -1 : 1) * DataRenderService.LabelLinePadding);
-                const chartPoint = outlineArc.centroid(<any>d);
+                textPoint[0] = textPoint[0] + ((midAngle(d) < Math.PI ? -1 : 1) * DataRenderService.LabelLinePadding);
+                const chartPoint = outlineArc.centroid(d);
                 chartPoint[0] *= DataRenderService.ChartLinePadding;
                 chartPoint[1] *= DataRenderService.ChartLinePadding;
 
-                return <any>[chartPoint, textPoint];
+                const result = [].concat(chartPoint).concat(textPoint)
```

Comment:
better to use a spread operator: `[...chartPoint, ...textPoint]`

Reply:
`chartPoint` and `textPoint` are already arrays, so logically `concat` works with them really good.
```TypeScript
const result = [].concat(chartPoint, textPoint);
```

Also, it's considerably faster at least in 10 times.